src/Nethermind/Nethermind.DataMarketplace.Initializers/NdmPlugin.cs

```diff
@@ -70,6 +80,13 @@ public Task InitRpcModules()
                 if (logger.IsInfo) logger.Info("Enabled JSON RPC Proxy for NDM.");
             }
 
+            if(_ndmInitializer == null)
+            {
+                throw new NullReferenceException("Ndm initializer is not created yet, can't start the rpc modules");
```

Comment:
Don't throw nullreference in your code, this is framework exception and will cause confusion. Throw InvalidOperationException

Reply:
Agreed or nul ref. I think tht ArgumentNullEx would be better to throw.

Reply:
It throws InvalidOperationException some lines earlier. its not really argumentof this method. ut its debatable I was to say either one, but seen that we already throw InvalidOperationException